plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt

```diff
@@ -266,7 +267,12 @@ private fun parseLockfile(podfileLock: File): LockfileData {
 
         val dependencies = node[entry]?.map { depNode ->
             val (depName, depVersion) = parseNameAndVersion(depNode.textValue())
-            depName.also { if (depVersion != null) versionForName[it] = depVersion }
+            depName.also {
+                // Versions on "second level" dependencies are actually constraints; only add it if is concrete.
```

Comment:
This makes me wonder if the previous PR which introduced parsing these constraints was actually the right thing to do. Could it be that each dependency which appears on the first level under  `DEPENDENCIES` has been resolved, but the second level are always really just constraints. Implying that if there is a dependency only on the second level, there was a constraint for it, which could not be resolved?   Might providing a pointer again to the rationale for the previous change?

Reply:
I've just read this section, and this sound like with a lockfile the version can really be pinned down exactly, See https://guides.cocoapods.org/using/pod-install-vs-update.html#using-exact-versions-in-the-podfile-is-not-enough.

I wonder what that last sentence with "properly use pod install vs. pod update." refers to. Any idea?

Reply:
It's a guess but maybe we must distinguish the following two cases.
Let's consider package `D` which is a dependency of package `A`:

1. If `D` is a subspec of `A` then use the version of `A` as the version for `D`
2. Otherwise assume / assert that there is a top-level entry for `D` with a version

Could this algorithm make sense?

Reply:
After doing some more research, I believe it's simpler than that. The (root) spec for AFNetworking over [here](https://github.com/AFNetworking/AFNetworking/blob/master/AFNetworking.podspec) lists no dependencies at all, only subspecs have [dependencies](https://guides.cocoapods.org/syntax/podspec.html#dependency). Still, the generated lockfile has [nested entries](https://github.com/oss-review-toolkit/ort/blob/4bbd26a8a4dde135054d4ceace5c90750882a3a0/plugins/package-managers/cocoapods/src/funTest/assets/projects/synthetic/external-sources/Podfile.lock#L3-L7) under AFNetworking. This tells me that these entries are indeed not dependencies, but pure constraint, say that *if* (and only if) one of the listed entries *also* appears as a dependency, then it must be in a version matching the constraint.

So for the AFNetworking case, AFNetworking/UIKit should not be part of the ORT dependency tree at all, but e.g. AFNetworking/NSURLSession should, as it also appears at the first level.

I'll make the necessary changes.